docs/source/10_deployment/12_dask.md

```diff
@@ -0,0 +1,313 @@
+            *to_build
+        )
+
+        # We also need any missing datasets that are required to run the
+        # `to_rerun` pipeline, including any chains of missing datasets.
+        unregistered_ds = pipeline.data_sets() - set(catalog.list())
+        # Some of the unregistered datasets could have been published to
+        # the scheduler in a previous run, so we need not recreate them.
+        missing_unregistered_ds = {
+            ds_name
+            for ds_name in unregistered_ds
+            if not self.create_default_data_set(ds_name).exists()
+        }
+        output_to_unregistered = pipeline.only_nodes_with_outputs(
+            *missing_unregistered_ds
+        )
+        input_from_unregistered = to_rerun.inputs() & missing_unregistered_ds
+        to_rerun += output_to_unregistered.to_outputs(*input_from_unregistered)
+
+        # We need to add any previously-published, unregistered datasets
+        # to the catalog passed to the `run` method, so that it does not
+        # think that the `to_rerun` pipeline's inputs are not satisfied.
+        catalog = catalog.shallow_copy()
+        for ds_name in unregistered_ds - missing_unregistered_ds:
+            catalog.add(ds_name, self.create_default_data_set(ds_name))
+
+        return self.run(to_rerun, catalog)
+```
+
+### Update CLI implementation
+
+You're nearly there! Before being able to use the new runner, update the `run()` function in your `cli.py` file to make sure the runner class is instantiated correctly:
+
+```python
+def run(tag, env, parallel, ...):
+    """Run the pipeline."""
+    if parallel and runner:
+        raise KedroCliError(
+            "Both --parallel and --runner options cannot be used together. "
+            "Please use either --parallel or --runner."
+        )
+    runner = runner or "SequentialRunner"
+    if parallel:
+        runner = "ParallelRunner"
+
+    tag = _get_values_as_tuple(tag) if tag else tag
+    node_names = _get_values_as_tuple(node_names) if node_names else node_names
+
+    with KedroSession.create(env=env, extra_params=params) as session:
+        context = session.load_context()
+        runner_instance = _instantiate_runner(runner, is_async, context)
+        session.run(
+            tags=tag,
+            runner=runner_instance,
+            node_names=node_names,
+            from_nodes=from_nodes,
+            to_nodes=to_nodes,
+            from_inputs=from_inputs,
+            to_outputs=to_outputs,
+            load_versions=load_version,
+            pipeline_name=pipeline,
+        )
+```
+
+where the helper function `_instantiate_runner()` looks like this:
+
+```python
+def _instantiate_runner(runner, is_async, project_context):
+    runner_class = load_obj(runner, "kedro.runner")
+    runner_kwargs = dict(is_async=is_async)
+
+    if runner.endswith("DaskRunner"):
+        client_args = project_context.params.get("dask_client") or {}
```

Comment:
Wouldn't this be 
```suggestion
        client_args = project_context.params.get("params.dask_client") or {}
```
?

Reply:
Don't think so:

```
(kedro-dask-example) WASH-178551-C02X31K9JHD4:kedro-dask-example deepyaman_datta$ kedro run --env dask --runner kedro_dask_example.runner.DaskRunner
2022-02-24 22:03:44,996 - kedro.framework.cli.hooks.manager - INFO - Registered CLI hooks from 1 installed plugin(s): kedro-telemetry-0.1.3
2022-02-24 22:03:45,179 - kedro_telemetry.plugin - INFO - You have opted into product usage analytics.
2022-02-24 22:03:45,492 - kedro.framework.session.store - INFO - `read()` not implemented for `BaseSessionStore`. Assuming empty store.
> /Users/deepyaman_datta/kedro-dask-example/src/kedro_dask_example/cli.py(61)_instantiate_runner()
-> client_args = project_context.params.get("dask_client") or {}
(Pdb) project_context.params
{'example_test_data_ratio': 0.2, 'example_num_train_iter': 10000, 'example_learning_rate': 0.01, 'dask_client': {'address': '127.0.0.1:8786'}}
(Pdb) 
```

This is the same way I've been running github.com/deepyaman/kedro-dask-example.

Reply:
Oh interesting, good to be reminded of that. 😂  Thanks for checking!